pyqtgraph/functions.py

```diff
@@ -997,7 +997,28 @@ def solveBilinearTransform(points1, points2):
         matrix[i] = numpy.linalg.solve(A, B[:,i])  ## solve Ax = B; x is one row of the desired transformation matrix
     
     return matrix
-    
+
+
+def clip_array(arr, vmin, vmax, out=None):
+    # replacement for np.clip due to regression in
+    # performance since numpy 1.17
+    # https://github.com/numpy/numpy/issues/14281
+
+    if out is None:
+        out = np.empty_like(arr)
+
+    if vmin is not None:
+        arr = np.core.umath.maximum(arr, vmin, out=out)
+    if vmax is not None:
+        arr = np.core.umath.minimum(arr, vmax, out=out)
```

Comment:
Doesn't this always clip the iput array, even if out is given?
I need to leave the input array intact when clipping only the displayed data in PlotDataItem.